modules/ROOT/pages/zero-to-hero.adoc

```diff
@@ -0,0 +1,294 @@
+key. We will use this later during verification. This can, and should, be distributed freely to
+anyone that may want to verify the image. `cosign.key` is the private key. This is used for signing
+content. Treat this like the password you provided. No one other than the entity responsible for
+singing and attestating images should have access to this.
+
+== Signing the Image
+
+We use cosign to sign our image.
+
+[.console-input]
+[source, bash]
+----
+cosign sign --key cosign.key $REPOSITORY:latest
+----
+
+Enter the key-pair password and `y` to upload a transaction record to
+https://docs.sigstore.dev/logging/overview/[Rekor].
+
+== Attesting the Image
+
+Next, let's use cosign again to associate a SLSA Provenance attestion to the image.
+
+Usually, the build system responsible for building the container image is also responsible for
+providing the SLSA Provenance attestation. Here, we will use an example one instead.
+
+First, create the content of the SLSA Provenance:
+
+[.console-input]
+[source, bash]
+----
+echo '{
+  "builder": {
+    "id": "https://localhost/dummy-id"
+  },
+  "buildType": "https://localhost/dummy-type",
+  "invocation": {},
+  "buildConfig": {},
+  "metadata": {
+    "buildStartedOn": "2023-09-25T16:26:44Z",
+    "buildFinishedOn": "2023-09-25T16:28:59Z",
+    "completeness": {
+      "parameters": false,
+      "environment": false,
+      "materials": false
+    },
+    "reproducible": false
+  },
+  "materials": []
+}
+' > predicate.json
+----
+
+Now, create an attestation and attach it to the container image.
+
+[.console-input]
+[source, bash]
+----
+cosign attest --predicate predicate.json --type slsaprovenance --key cosign.key "$REPOSITORY:latest"
+----
+
+As done while signing the image, enter the key-pair password and `y` to upload a transaction record
+to Rekor.
+
+== Quick Check
+
+At this point, the command `cosign tree` should detect the image has one signature and one
+attestation.
+
+[.console-input]
+[source, bash]
+----
+$ cosign tree "$REPOSITORY:latest"
+📦 Supply Chain Security Related artifacts for an image: localhost:5000/ec-zero-to-hero:latest
+└── 💾 Attestations for an image tag: localhost:5000/ec-zero-to-hero:sha256-b5430d3d447434c795a508036e5046e41c009039be5b3f656f121c2426500d1e.att
+   └── 🍒 sha256:ad5ee60aedd8ada59a559cd2e4b83ccc9bd1aaa97a8b48ba151d728afec31bbc
+└── 🔐 Signatures for an image tag: localhost:5000/ec-zero-to-hero:sha256-b5430d3d447434c795a508036e5046e41c009039be5b3f656f121c2426500d1e.sig
+   └── 🍒 sha256:09dc4217ce34b7db29acb9e914067db2092e8dfbf4c7603857114d9b2ac7f0ab
+
+----
+
+We are ready to verify the image with the Enterprise Contract CLI!
+
+== Basic Verification
+
+The most basic verification that can be done with the EC cli is to verify the image has a signature
+and a SLSA Provenance attestation matching a given public key.
+
+[.console-input]
+[source, bash]
+----
+ec validate image --public-key cosign.pub --image "$REPOSITORY:latest" --policy ''
```

Comment:
Kinda ugly that we need to provide `''` for `--policy`, should we change the [default](https://github.com/enterprise-contract/ec-cli/blob/439d1cc00a425255cb08a4a0e54d05ebdb0a8a9c/cmd/validate/image.go#L70)? Not sure if anyone relies on that. We could also have a `$HOME/.ec/config` with the current default in the image used in RHTAP...

Reply:
Agree. Filed https://github.com/enterprise-contract/ec-cli/issues/1039